feature: use new source for binaries and update docs (#297)
codacy/codacy-coverage-reporter: docs/alternative-ways-of-running-coverage-reporter.md

If you prefer, you can manually download and run the native `codacy-coverage-reporter` binary, either for the latest version or a specific one.

> You can use the scripts below to automatically check for the latest version of the binaries, download the binaries from either our public S3 or GitHub, and run them.

Reviewer: Small tweak to make the two alternatives clearer:

```suggestion
You can use the scripts below to automatically check for the latest version of the binaries, download the binaries from either Codacy's public Amazon S3 or GitHub, and run them.
```

Reviewer 2: I think we shouldn't expose here that the artifacts are stored in S3. It is an implementation detail.
It can be "our public store", "our public link" or whatever else.

Reviewer: Yes, I like @lolgab's suggestion even better! :+1:

> You can use the scripts below to automatically check for the latest version of the binaries, download the binaries from either our public S3 or GitHub, and run them.

Author: ```suggestion
You can use the scripts below to automatically check for the latest version of the binaries, download the binaries from either our public link or GitHub, and run them.
```

Use the Java binary to run Codacy Coverage reporter on other platforms, such as Linux x86, macOS, Windows, etc.

> You can use the scripts below to automatically check for the latest version of the Java binaries, download the binaries from either our public S3 or GitHub, and run them.

Reviewer: Same suggestion as above:

```suggestion
You can use the scripts below to automatically check for the latest version of the Java binaries, download the binaries from either Codacy's public Amazon S3 or GitHub, and run them.
```

> You can use the scripts below to automatically check for the latest version of the Java binaries, download the binaries from either our public S3 or GitHub, and run them.

Author: ```suggestion
You can use the scripts below to automatically check for the latest version of the Java binaries, download the binaries from either our public store or GitHub, and run them.
```

> -   Using Codacy's public S3:

Author: ```suggestion
-   Using Codacy's public store:
```

    ```bash

>     LATEST_VERSION="$(curl -Ls https://artifacts.codacy.com/binaries/codacy-coverage-reporter/latest)"
>     curl -Ls -o codacy-coverage-reporter-assembly.jar "https://artifacts.codacy.com/binaries/codacy-coverage-reporter/${LATEST_VERSION}/codacy-coverage-reporter-assembly.jar"

Author: ```suggestion
    LATEST_VERSION="$(curl -Ls https://artifacts.codacy.com/bin/codacy-coverage-reporter/latest)"
    curl -Ls -o codacy-coverage-reporter-assembly.jar "https://artifacts.codacy.com/bin/codacy-coverage-reporter/${LATEST_VERSION}/codacy-coverage-reporter-assembly.jar"
```
